api/src/services/query.service.ts

```diff
@@ -24,32 +51,34 @@ export class QueryService {
   }
 
   private async postgresqlQuery(key: string, sql: string): Promise<object[]> {
-    const sourceConfig = await DataSourceService.getByTypeKey('postgresql', key);
-    const configuration = configureDatabaseSource('postgresql', sourceConfig.config);
-    const source = new DataSource(configuration);
+    let source = QueryService.getDBConnection('postgresql', key);
+    if (!source) {
+      const sourceConfig = await DataSourceService.getByTypeKey('postgresql', key);
+      const configuration = configureDatabaseSource('postgresql', sourceConfig.config);
+      source = new DataSource(configuration);
+      QueryService.addDBConnection('postgresql', key, source);
+    }
     const result = await this.executeQuery(source, sql);
     return result;
   }
 
   private async mysqlQuery(key: string, sql: string): Promise<object[]> {
-    const sourceConfig = await DataSourceService.getByTypeKey('mysql', key);
-    const configuration = configureDatabaseSource('mysql', sourceConfig.config);
-    const source = new DataSource(configuration);
+    let source = QueryService.getDBConnection('mysql', key);
+    if (!source) {
+      const sourceConfig = await DataSourceService.getByTypeKey('mysql', key);
+      const configuration = configureDatabaseSource('mysql', sourceConfig.config);
+      source = new DataSource(configuration);
+      QueryService.addDBConnection('mysql', key, source);
+    }
     const result = await this.executeQuery(source, sql);
     return result;
   }
 
   private async executeQuery(source: DataSource, sql): Promise<any> {
-    try {
+    if (!source.isInitialized) {
       await source.initialize();
```

Comment:
How about moving the `source.initialize()` to the line after `source = new DataSource(configuration);`?

IIUR, this method should be called once on application bootstrap.

Reply:
Modified in latest commit

Reply:
LGTM